engine/overworld/weather.asm

```diff
@@ -0,0 +1,778 @@
+DoOverworldWeather:
+	push de
+	push hl
+	push bc
+	ld hl, wWeatherFlags
+	bit OW_WEATHER_DISABLED_F, [hl]
+	jr nz, .done
+	ld a, [wOverworldWeatherDelay]
+	and %1 ; 30 fps
+	jr z, .done
+	ld a, [wOverworldWeatherCooldown]
+	and a
+	jr z, .no_cooldown
+	dec a
+	ld [wOverworldWeatherCooldown], a
+.no_cooldown
+	ldh a, [hUsedOAMIndex]
+	cpl
+	add (SPRITEOAMSTRUCT_LENGTH * NUM_SPRITE_OAM_STRUCTS) + 1
+	ld hl, hUsedWeatherSpriteIndex
+	cp [hl]
+	jr nc, .ok
+	add -SPRITEOAMSTRUCT_LENGTH
+	ldh [hUsedWeatherSpriteIndex], a
+.ok
+	ld a, [wOverworldWeatherCooldown]
+	and a
+	jr nz, .on_cooldown
+	ld a, [wCurrentWeather]
+	dec a ; OW_WEATHER_RAIN
+	call z, DoOverworldRain
+	ld a, [wCurrentWeather]
```

Comment:
I feel like there has to be a better way of doing this than reloading the weather variable and checking for each kind of weather, when we know it will only be one (or none if zero). How about a jumptable, with `DoNothing` for no weather?